src/cloud-api-adaptor/libvirt/config_libvirt.sh

```diff
@@ -86,8 +86,6 @@ installKcli() {
             # export PATH="$PATH:$HOME/.local/bin"
             pipx install kcli==${kcli_version}
             pipx ensurepath
-            # Reload shell so that pipx install PATH is available
```

Comment:
Running `exec $SHEL` here on gha will definitively stop here and do not run important setup steps (like creating the ssh key-pair).

Sorry, I missed the first discussion about that exec... would make sense to append `$HOME/.local/bin` to system's PATH if in gha action as shown in `https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path`

Reply:
I did that in the previous PR: https://github.com/confidential-containers/cloud-api-adaptor/blob/8f534e3e27f81d4794d8bd548a276a67fad99773/.github/workflows/e2e_libvirt.yaml#L125

Reply:
but yes, @wainersm is right. if we pivot to a new shell mid-script, the rest of the script will be discarded in any case (locally or on the CI). There are a gazillion ways to update the path, but we should probably focus on unblocking the CI.

one of those options would be to have pipx installed in an outside step, the next steps starts a new shell and should have the path set. 

```yaml
name: install pipx
if: ... # onyl on ubuntu
run: |     
  sudo apt-get update
  sudo DEBIAN_FRONTEND=noninteractive apt-get  install pipx -y
  pipx ensurepath
name: run config-libvirt
run: ...  
```

Reply:
I definitely understand the approach is probably nicer, but I would prefer to not move pipx in a separate step as I'd like to move more towards moving things from the workflow into shell scripts for easier PR testing than the opposite.

I've tried adding a new commit with `"${CI}" != "true"` which should mean we don't run the reload at all on GHA. Hopefully that is sufficient to get this PR merged and unblock the CI and I'm happy to discuss further improvements once we are back green again.

Reply:
I think the original approach of opening a new shell at the end of the script should at least do no harm. me, as an interactive user of a script, won't like this, but that's maybe idiosyncrasy